sending/src/main/java/org/kin/sending/view/RecipientContactsAdapter.java

```diff
@@ -0,0 +1,93 @@
+package org.kin.sending.view;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.kin.sending.R;
+import org.kin.sendkin.core.model.RecipientContact;
+import org.kin.sendkin.core.store.RecipientContactsRepo;
+import org.kin.sendkin.core.view.Utils;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+public class RecipientContactsAdapter extends RecyclerView.Adapter<RecipientContactsAdapter.RecipientContactView> {
+
+    public interface RecipientContactTouchListener {
+        void onEditContact(@NonNull UUID id);
+
+        void onContactClicked(@NonNull UUID id);
+    }
+
+    private ArrayList<RecipientContact> contacts;
+    private RecipientContactsRepo recipientContactsRepo;
+    private RecipientContactTouchListener recipientContactTouchListener;
+
+    public RecipientContactsAdapter(RecipientContactsRepo recipientContactsRepo, RecipientContactTouchListener recipientContactTouchListener) {
+        this.recipientContactsRepo = recipientContactsRepo;
+        this.recipientContactTouchListener = recipientContactTouchListener;
+        refreshData();
+    }
+
+    @Override
+    public RecipientContactView onCreateViewHolder(ViewGroup parent,
+                                                   int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.contact_item, parent, false);
+        return new RecipientContactView(view);
+    }
+
+    @Override
+    public void onBindViewHolder(RecipientContactView holder, int position) {
+        final RecipientContact recipientContact = contacts.get(position);
+        holder.bind(recipientContact);
+        holder.edit.setOnClickListener(new View.OnClickListener() {
```

Comment:
It is better to setOnClickListener in the ViewHolder instead of in onBindViewHolder(), that is because onBindViewHolder() is called for each and every item and setting the click listener is an unnecessary option to repeat when you can call it once in your ViewHolder constructor. Then, if your click responds depends on the position of the item clicked, you can simply call getAdapterPosition() from within the ViewHolder.

Reply:
@drEfrati great point!!!